aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -295,6 +303,55 @@ impl<
         );
     }
 
+    pub fn remove_v2(
+        &self,
+        group_key: &K,
+        txn_idx: TxnIndex,
+        tags: HashSet<T>,
+    ) -> Result<BTreeSet<(TxnIndex, Incarnation)>, PanicError> {
+        let mut ret = BTreeSet::new();
+        self.remove_impl::<true>(group_key, txn_idx, tags, &mut ret);
+
+        let mut group_sizes = self
+            .group_sizes
+            .get_mut(group_key)
+            .expect("Path must exist");
+        let removed_size = group_sizes
+            .size_entries
+            .remove(&ShiftedTxnIndex::new(txn_idx));
+
+        match removed_size {
+            Some(size_entry) => {
+                // Handle removed dependencies.
+                let mut removed_deps: BTreeSet<(TxnIndex, Incarnation)> =
+                    std::mem::take(&mut size_entry.value.1.lock());
+                if let Some((_, next_entry)) = group_sizes
+                    .size_entries
+                    .range(..=ShiftedTxnIndex::new(txn_idx))
+                    .next_back()
+                {
+                    if next_entry.value.0 == size_entry.value.0 {
+                        next_entry
+                            .value
+                            .1
+                            .lock()
+                            .extend(std::mem::take(&mut removed_deps));
```

Comment:
having a comment here helps explain about what this does - merge the dependency if the value stays the same or return it for invalidation. I was looking at L342 and wondering what value it'll get.

it seems we don't need this `ret` too? just return removed_deps seems enough?

Reply:
@zekun000  Do you mean having if { extend(removed_deps) } else { ret.extend(removed_deps)}?